tests/test_outlier.py

```diff
@@ -0,0 +1,552 @@
+        OOD.fit(pred_probs=pred_probs, params={"strange_param": -1})
+
+    # testing OOD param in fit for outlier detection
+    mixed_params_dict = {"k": 10, "adjust_pred_probs": True}
+    try:
+        OOD.fit(features=features, params=mixed_params_dict)
+    except Exception as e:
+        assert "with features, passed in params dict can only contain" in str(e)
+        with pytest.raises(ValueError) as e:
+            OOD.fit(features=features, params=mixed_params_dict)
+
+    # testing outlier param in fit for OOD detection
+    try:
+        OOD.fit(pred_probs=pred_probs, params=mixed_params_dict)
+    except Exception as e:
+        assert "with pred_probs, passed in params dict can only contain" in str(e)
+        with pytest.raises(ValueError) as e:
+            OOD.fit(features=features, params=mixed_params_dict)
+
+    #### SCORE wrong info
+
+    # TESTING: calling score before any fitting
+    try:
+        OOD.score(features=features)
+    except Exception as e:
+        assert "OOD Object needs to be fit on features first." in str(e)
+        with pytest.raises(ValueError) as e:
+            OOD.score(features=features)
+
+    try:
+        OOD.score(pred_probs=pred_probs)
+    except Exception as e:
+        assert "OOD Object needs to be fit on features first." in str(e)
+        with pytest.raises(ValueError) as e:
+            OOD.score(pred_probs=pred_probs)
+
+    # TESTING: calling scoring with opposite fitting
+    OOD_outlier = OutOfDistribution()
+    OOD_outlier.fit(features=features)
+    try:
+        OOD_outlier.score(pred_probs=pred_probs)
+    except Exception as e:
+        assert "OOD Object needs to be fit on features first." in str(e)
+        with pytest.raises(ValueError) as e:
+            OOD_outlier.score(pred_probs=pred_probs)
+
+    OOD_ood = OutOfDistribution()
+    OOD_ood.fit(pred_probs=pred_probs, labels=labels)
+    try:
+        OOD_ood.score(features=features)
+    except Exception as e:
+        assert "OOD Object needs to be fit on features first." in str(e)
+        with pytest.raises(ValueError) as e:
+            OOD_ood.score(features=features)
+
+    # TESTING: calling ood after fitting but without adjust_pred_probs
+    OOD_ood = OutOfDistribution()
+
+    #  This should throw warning since no confident_thresholds calculated
+    OOD_ood.fit(pred_probs=pred_probs, params={"adjust_pred_probs": False})
+    OOD_ood.score(pred_probs=pred_probs)  # This should be ok since we are not adjusting
+
+
+def test_class_params_logic():
+    features = data["X_train"]
+    pred_probs = data["pred_probs"]
+
+    OOD = OutOfDistribution()
+
+    # TESTING: params dict is a copy
+    params_dict = {"k": 10, "t": 5}
+    OOD.fit(features=features, params=params_dict)
+    ood_params = OOD.params
+    params_dict = params_dict.update({"k": 20})
+    assert ood_params == OOD.params
+
+    # test calling functions with different params performs differently
+
```

Comment:
also test that you can set `labels=None` while providing pred_probs if `adjust_pred_probs=False`?